### PR TITLE
spotifyd: new package at 0.2.24

### PR DESCRIPTION
**Pull request by Jaycoobs**
## Description of package
Spotifyd allows third party Spotify clients to be able to stream music without relying on the official client

## Installed manifest of package

```
/var/db/kiss/installed/spotifyd/version
/var/db/kiss/installed/spotifyd/sources
/var/db/kiss/installed/spotifyd/manifest
/var/db/kiss/installed/spotifyd/depends
/var/db/kiss/installed/spotifyd/checksums
/var/db/kiss/installed/spotifyd/build
/var/db/kiss/installed/spotifyd/
/var/db/kiss/installed/
/var/db/kiss/
/var/db/
/var/
/usr/bin/spotifyd
/usr/bin/
/usr/
```

## New package

- [x] Latest upstream version.
- [x] I agree to maintain this package.
- [x] I agree to notify @dylanaraps if I can no longer maintain this package.

### CODE REVIEW

File: community/spotifyd/build [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/sh -e
+
+# Spotifyd won't compile with libressl versions past 3.0.2 :/
+(
+	cd libressl
+
+	./configure
+	make check
+	make DESTDIR="$PWD/../libressl302" install
+)
```
Comment (dylanaraps) [inline community/spotifyd/build:10]: What's the compilation error from latest libressl?

Comment (Jaycoobs) [reply community/spotifyd/build:10]: I run into this mess.

```
error: failed to run custom build command for `openssl-sys v0.9.53`

Caused by:
  process didn't exit successfully: `/root/.cache/kiss/build-6190/spotifyd/target/release/build/openssl-sys-a1768c18737b3939/build-script-main` (exit code: 101)
  --- stdout
  cargo:rustc-cfg=const_fn
  cargo:rerun-if-env-changed=X86_64_UNKNOWN_LINUX_MUSL_OPENSSL_LIB_DIR
  X86_64_UNKNOWN_LINUX_MUSL_OPENSSL_LIB_DIR unset
  cargo:rerun-if-env-changed=OPENSSL_LIB_DIR
  OPENSSL_LIB_DIR unset
  cargo:rerun-if-env-changed=X86_64_UNKNOWN_LINUX_MUSL_OPENSSL_INCLUDE_DIR
  X86_64_UNKNOWN_LINUX_MUSL_OPENSSL_INCLUDE_DIR unset
  cargo:rerun-if-env-changed=OPENSSL_INCLUDE_DIR
  OPENSSL_INCLUDE_DIR unset
  cargo:rerun-if-env-changed=X86_64_UNKNOWN_LINUX_MUSL_OPENSSL_DIR
  X86_64_UNKNOWN_LINUX_MUSL_OPENSSL_DIR unset
  cargo:rerun-if-env-changed=OPENSSL_DIR
  OPENSSL_DIR unset
  cargo:rustc-link-lib=ssl
  cargo:rustc-link-lib=crypto
  OPT_LEVEL = Some("3")
  TARGET = Some("x86_64-unknown-linux-musl")
  HOST = Some("x86_64-unknown-linux-musl")
  CC_x86_64-unknown-linux-musl = None
  CC_x86_64_unknown_linux_musl = None
  HOST_CC = None
  CC = None
  CFLAGS_x86_64-unknown-linux-musl = None
  CFLAGS_x86_64_unknown_linux_musl = None
  HOST_CFLAGS = None
  CFLAGS = Some("-march=x86-64 -mtune=generic -pipe -Os")
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  running: "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-march=x86-64" "-mtune=generic" "-pipe" "-Os" "-I" "/usr/include" "-E" "build/expando.c"
  exit code: 0
  cargo:rustc-cfg=osslconf="OPENSSL_NO_BUF_FREELISTS"
  cargo:rustc-cfg=osslconf="OPENSSL_NO_COMP"
  cargo:rustc-cfg=osslconf="OPENSSL_NO_KRB5"
  cargo:rustc-cfg=osslconf="OPENSSL_NO_PSK"
  cargo:rustc-cfg=osslconf="OPENSSL_NO_RFC3779"
  cargo:rustc-cfg=osslconf="OPENSSL_NO_SRP"
  cargo:rustc-cfg=osslconf="OPENSSL_NO_SSL3_METHOD"
  cargo:conf=OPENSSL_NO_BUF_FREELISTS,OPENSSL_NO_COMP,OPENSSL_NO_KRB5,OPENSSL_NO_PSK,OPENSSL_NO_RFC3779,OPENSSL_NO_SRP,OPENSSL_NO_SSL3_METHOD
  cargo:rustc-cfg=libressl
  cargo:rustc-cfg=libressl251
  cargo:rustc-cfg=libressl261
  cargo:rustc-cfg=libressl270
  cargo:rustc-cfg=libressl271
  cargo:rustc-cfg=libressl273
  cargo:rustc-cfg=libressl280
  cargo:rustc-cfg=libressl281
  cargo:rustc-cfg=libressl291
  cargo:libressl_version_number=3020100f

  --- stderr
  thread 'main' panicked at '

  This crate is only compatible with OpenSSL 1.0.1 through 1.1.1, or LibreSSL 2.5
  through 3.0.x, but a different version of OpenSSL was found. The build is now aborting
  due to this version mismatch.

  ', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/openssl-sys-0.9.53/build/main.rs:243:5
warning: build failed, waiting for other jobs to finish...
error: build failed
-> spotifyd Build failed
-> spotifyd Log stored to /root/.cache/kiss/logs/spotifyd-2020-09-07-08:12-6190
Terminated
```

Comment (dylanaraps) [reply community/spotifyd/build:10]: This is due to the openssl-sys crate hardcoding libressl versions. We patch this crate in the rust package for the same error message. To do the same here we'll have to vendor the crate. I'll see what I can come up with and send a patch your way. :+1:

**Comment by dylanaraps** [comment]
I have spotifyd working with the latest libressl. Here are the changed files:

https://gist.github.com/dylanaraps/128323214fe8c85fc8e8f978097f3b37

This works by vendoring the openssl-sys crate which then allows us to patch it to support newer libressl versions.

**Pull request merged**